### DISCUSSION TITLE
Welcome to Arduino-Atmel-sPWM Discussions!

### DISCUSSION THREAD

**Discussion by Irev-Dev** [Irev-Dev/Arduino-Atmel-sPWM#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place for general discussions. Ways you can use this space:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Ask for help. (the more specific your question the easier it will be to offer help

**Comment by gjvengu** [comment]
Hello Sir

**Comment by Ronald101101** [comment]
Can i generate 1khz sine wave using this if so how?

**Reply by Irev-Dev** [reply]
From memory the switching frequency is about 20khz, so it would only be 20 pulses per cycle which is probably too few/course.

Have you watched the video? https://www.youtube.com/watch?v=Pf2Jx5LMXH0
Might give you a better idea of what would need to change to get a decent 1khz wave.

Bare in mind arduinos have 16mhz crystal, which is not terribly fast.

**Comment by T31328** [comment]
Hello Sir
I am having an issue understanding the code for switching frequency and how to increase the switching frequency. Kindly help me to understand the formula given in the code for the period and its relation with Switching frequency.

**Reply by Irev-Dev** [reply]
Sorry mate, your question is very vague, hard to know where to start. Is there anything specific you don't understand about the code?

And for completeness's sake, Are you aware there are videos on youtube explaining the code?
https://www.youtube.com/watch?v=Pf2Jx5LMXH0

**Reply by T31328** [reply]
Yes, Sir, I have watched the video. My question is simply related to the Switching frequency how can I increase the switching frequency by using this code? Secondly, there is a period calculated in this code what is that period related with the switching frequency

**Comment by ProgArtDaVinci** [comment]
The last code of the interrupt service program in the article does not work correctly. After lookUp[99]-A, lookUp[0]-A is performed. And then the channel is switched to B and lookUp[1]-B is performed. LookUp[0]-B is skipped. Further, such a defect also occurs with channel A. And this can be seen on the oscilloscope in practice. Redundant signal.
I suggest a simpler code:
```
ISR( TIMER1_OVF_vect ) {
	static int num;
	if( num >= SinDivisions/2 )
	{
		TCCR1A ^= 0b10100000;	// Toggle connect and disconnect of compare output A and B.
		num = 0;                // Reset num
	}
	OCR1A = lookUp[ num ];
	OCR1B = lookUp[ num ];
	num++;
}
```
